packages/winpanda/extra/src/winpanda/core/cmdconf.py

```diff
@@ -354,30 +319,120 @@ def get_dcos_conf(self):
-                template = self.load_dcos_conf_templete(template_fpath)
+                template = self.load_dcos_conf_template(template_fpath)
         except Exception as e:
             if not isinstance(e, cr_exc.RCError):
                 raise cr_exc.RCExtractError(
-                    f'DC/OS aggregated config: {type(e).__name__}: {e}'
+                    f'DC/OS aggregated config package: {dcoscfg_pkg_fpath}:'
+                    f' {type(e).__name__}: {e}'
                 )
             else:
                 raise
         else:
+            LOG.debug(f'{self.msg_src}: DC/OS aggregated config package:'
+                      f' {dcoscfg_pkg_fpath}: Preprocess: OK')
             return {'template': template, 'values': values}
         finally:
             dcoscfg_pkg_fpath.unlink()
 
+    def get_dstor_dcoscfgpkg_path(self, dstor_root_url, dstor_lpi_path):
+        """Retrieve the Linux Package Index (LPI) object from the DC/OS
+        distribution storage and discover a relative URL to the DC/OS
+        aggregated configuration package.
+        LPI is expected to be a JSON-formatted file containing descriptors for
+        DC/OS distribution packages:
+
+        {
+            "<pkg-name>":{
+                "filename":"<base-path>/<pkg-name>--<pkg-version>.tar.xz",
+                "id":"<pkg-name>--<pkg-version>"
+            },
+            ...
+        }
+
+        :param dstor_root_url:         str, DC/OS distribution storage root URL
+        :param dstor_lpi_path:         str, URL path to the DC/OS Linux package
+                                       index object at the DC/OS distribution
+                                       storage
+        :return dstor_dcoscfgpkg_path: str, URL path to the DC/OS aggregated
+                                       config package at the DC/OS distribution
+                                       storage
+        """
+        dcos_conf_pkg_name = 'dcos-config-win'
+
+        # Linux package index direct URL
+        lpi_url = posixpath.join(dstor_root_url, dstor_lpi_path)
+        lpi_fname = Path(dstor_lpi_path).name
+
+        try:
+            cm_utl.download(lpi_url, str(self.inst_storage.tmp_dpath))
+            LOG.debug(f'{self.msg_src}: DC/OS Linux package index: Download:'
+                      f' {lpi_fname}: {lpi_url}')
+        except Exception as e:
+            raise cr_exc.RCDownloadError(
+                f'DC/OS Linux package index: {lpi_fname}:'
+                f' {lpi_url}: {type(e).__name__}: {e}'
+            ) from e
+
+        lpi_fpath = self.inst_storage.tmp_dpath.joinpath(lpi_fname)
+
+        try:
+            lpi = cr_utl.rc_load_json(lpi_fpath,
+                                      emheading='DC/OS Linux package index')
+
+            if not isinstance(lpi, dict):
+                raise cr_exc.RCInvalidError(
+                    f'DC/OS Linux package index: {lpi_fpath}: Invalid'
+                    f' structure: {lpi}'
```

Comment:
We don't need to display the contents of this large file (`{lpi}`) in the error message. It would be reasonable to provide the `{lpi_url}` to give an indication of where we obtained the package index.

We can also leave out the `{lpi_fpath}` in all these messages. That file is deleted at the end of this function, so we should use the URL in all cases.

Reply:
Fixed in the last PR update.